### DISCUSSION TITLE
Libical - on the road to 4.x

### DISCUSSION THREAD

**Discussion by winterz** [libical/libical#745] [General]
I'd really like to finally start moving toward a 4.0 release (from the master branch)

Here's how you can help:

- Review the [issues](https://github.com/libical/libical/issues)
  - add a 4.0 milestone to all issues that need to be resolved before the 4.0 release
  - anything with a 3.1 milestone should be moved to the 4.0 milestone (or the milestone removed, where applicable)
  - add missing issues we neglected to add to the issue tracker
- Review the [pull requests](https://github.com/libical/libical/pulls)
  - update (and merge) any pull request as needed for the 4.0 release
  - close pull requests that are no longer relevant
- Documentation
  - we need a "What's New in 4.x" and a "Porting to 4.x" - yes we have the ReleaseNotes, but the list is so long in there and a short guide on how the API has changed and how to adjust your client code is needed
  - any other documentation improvements (a "book" would be nice with various .md files we put together using something like [mdbook](https://github.com/rust-lang/mdBook)
  - Fix all known security fuzzer issues (would be great to fix **all** fuzzer issues) (see https://oss-fuzz.com/testcases?project=libical)
- github CI
  - add more platforms (modern, like mac-14)
  - add static analysis on a nightly (cppcheck, clang-tidy, etc)
  - add documentation generation
- anything else I'm forgetting

**Comment by otao** [comment]
Good Morning. 

I just wanted to kindly ask if there’s any update on when the v4.x release might be available? 

Thanks!

**Comment by winterz** [comment]
Good question.
Progress has stalled for the past couple months as  I've been concentrating on other open source projects.

Sorry I can't give you an ETA.  not soon.

I'm still looking for help with the todo list in my original post.

**Comment by winterz** [comment]
The current set of milestone issues can be seen at https://github.com/libical/libical/milestone/14

some are release blockers; some can be done later.

feel free to start working on any of those -- don't forget to assign yourself.
hi priority are API changes

**Comment by winterz** [comment]
I think we're about ready to start the release process for 4.0

Looking at the issue tracker I don't see any blockers so I think we can start moving ahead now.

Basically this is a heads-up that I'll be working on the 4.0 release starting soon.

If you know of any blockers or other issues that we need to get fixed before 4.0 lands please let me know as soon as possible.
I know we're still negotiating what to do about libicalvcard.

-Allen
